app/charts/index.ts

```diff
@@ -2180,35 +2188,53 @@ const chartConfigsAdjusters: ChartConfigsAdjusters = {
-          lineUnitConversion = oldChartConfig.fields.y.rightAxisUnitConversion;
-        } else if (isComboLineSingleConfig(oldChartConfig)) {
-          leftMeasure = getMeasure(oldChartConfig.fields.y.componentIds[0]);
-          columnUnitConversion = oldChartConfig.fields.y.unitConversion;
-        } else if (
-          isAreaConfig(oldChartConfig) ||
-          isColumnConfig(oldChartConfig) ||
-          isLineConfig(oldChartConfig) ||
-          isPieConfig(oldChartConfig) ||
-          isScatterPlotConfig(oldChartConfig)
-        ) {
-          leftMeasure = getMeasure(oldChartConfig.fields.y.componentId);
-          columnUnitConversion = oldChartConfig.fields.y.unitConversion;
-        } else if (isMapConfig(oldChartConfig)) {
-          const { areaLayer, symbolLayer } = oldChartConfig.fields;
-          const leftAxisId =
-            areaLayer?.color.componentId ?? symbolLayer?.measureId;
-
-          if (leftAxisId) {
+        switch (oldChartConfig.chartType) {
+          case "comboLineDual": {
+            const leftAxisId = oldChartConfig.fields.y.leftAxisComponentId;
             leftMeasure = getMeasure(leftAxisId);
+            rightAxisMeasureId = oldChartConfig.fields.y.rightAxisComponentId;
+            columnUnitConversion =
+              oldChartConfig.fields.y.leftAxisUnitConversion;
+            lineUnitConversion =
+              oldChartConfig.fields.y.rightAxisUnitConversion;
+            break;
+          }
+          case "comboLineSingle": {
+            leftMeasure = getMeasure(oldChartConfig.fields.y.componentIds[0]);
+            columnUnitConversion = oldChartConfig.fields.y.unitConversion;
+            break;
+          }
+          case "area":
+          case "column":
+          case "line":
+          case "pie":
+          case "scatterplot": {
+            leftMeasure = getMeasure(oldChartConfig.fields.y.componentId);
+            columnUnitConversion = oldChartConfig.fields.y.unitConversion;
+            break;
+          }
+          case "map": {
+            const { areaLayer, symbolLayer } = oldChartConfig.fields;
+            const leftAxisId =
+              areaLayer?.color.componentId ?? symbolLayer?.measureId;
+
+            if (leftAxisId) {
+              leftMeasure = getMeasure(leftAxisId);
+            }
+
+            break;
+          }
+          case "bar": {
+            leftMeasure = getMeasure(oldChartConfig.fields.x.componentId);
+            columnUnitConversion = oldChartConfig.fields.x.unitConversion;
+            break;
           }
-        } else if (isBarConfig(oldChartConfig)) {
-          leftMeasure = getMeasure(oldChartConfig.fields.x.componentId);
-          columnUnitConversion = oldChartConfig.fields.x.unitConversion;
+          case "comboLineColumn":
+          case "table":
+            break;
+          default:
+            const _exhaustiveCheck: never = oldChartConfig;
+            return _exhaustiveCheck;
```

Comment:
I am scanning here but I am quite impressed by the extent you had to check here.

I wonder now if the conversion of unit should not happen at the data layer = converting the observations directly and keeping the unitConversion config at chart level, indexed by component, instead of it being an UI thing that is bound to fields (which needs to be carried over when converting the chart). I think the approach would be simpler but maybe it has downfalls I do not see.